src/main/scala/jigg/util/JSONUtil.scala

```diff
@@ -116,6 +120,64 @@ object JSONUtil {
   }
 
   def writeToJSON[T <: Node](x: T, os: BufferedWriter): Unit = {
-    os.write(toJSON(x).toString)
+    os.write(toJSON(x))
+  }
+
+  def parseJSON(json: String): JValue = parse(json)
+
+  def loadJSONFile(name: String): JValue = {
+    val sb = new StringBuilder
+    for (line <- Source.fromFile(name).getLines){
+      sb.append(line)
+    }
+    parse(sb.toString)
+  }
+
+  def toXML(json: JValue): Node = {
+    implicit val formats = DefaultFormats
+    val jsonList = json.extract[Map[String, Any]].toList
+    generateXML(jsonList)
+  }
+
+  private def generateXML(x:List[(String, Any)]): Node = {
+    var tagString = ""
+    var textString = ""
+    val attributeList = new ArrayBuffer[(String, String)]
+    val nodeSeq = new ArrayBuffer[Node]
+    var hasChildFlag = false
+    var hasTextFlag = false
+
+    for (i <- x){
```

Comment:
This style seems very procedural. Following code might be clearer.

```scala
    // Type of x is Map[String, Any]
    val tagString = x get ".tag" // Option[Any], but this should always exist.
    val textString = x get "text" // This might be None.
    val children: List[Node] =
      x.get(".child").map { _.asInstanceOf[List[Map[String, Any]]]
        .map(generateXML) } getOrElse List()

    val attrs: Map[String, String] =
      x.filter { case (k, v) => k != ".tag" && k != "text" && k != ".child" }
        .map { case (k, v) => (k, v.toString) }

    val node = textString match {
      case Some(text) => <xml>{text}</xml>
      case None => <xml/>
    }
    val tagChanged = node.copy(label = tagString.get.toString)
    val childAdded = XMLUtil.addChild(tagChanged, children) // do nothing when children is empty
    XMLUtil.addAttributes(childAdded, attrs) // the same
```

This code does not use mutable objects like `var` or `ArrayBuffer` to make the code intuitive, though I've not tested the behavior.